diff --git a/Modelling/dataset/tensorflow.py b/Modelling/dataset/tensorflow.py
index 8870d43..8e192cc 100644
--- a/Modelling/dataset/tensorflow.py
+++ b/Modelling/dataset/tensorflow.py
@@ -2,118 +2,33 @@ from Modelling import *
 
 
 class TensorFlow_Data_Loader:
-    # def __init__(
-    #     self,
-    #     data: pd.DataFrame = pd.read_csv("./Modelling/dataset/data/train.csv"),
-    #     test: pd.DataFrame = pd.read_csv("./Modelling/dataset/data/test.csv"),
-    # ):
-    #     self.data = data
-    #     self.data = self.data.sample(frac=1)
-    #     self.test = test
-    #     self.X = np.array(self.data["text"])
-    #     self.y = np.array(self.data["target"])
-    #     self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(
-    #         self.X, self.y, test_size=0.0625, shuffle=True
-    #     )
-
-    # def create(self):
-    #     return self.X_train, self.X_test, self.y_train, self.y_test
-
-    # def create_test(self):
-    #     return np.array(self.test["text"])
-
-    # def create_submission(self, model):
-    #     preds = model.predict(self.create_test())
-    #     print(preds)
-    #     ids = self.test["id"]
-    #     submission = {"id": [], "target": []}
-    #     for pred, id in tqdm(zip(preds, ids)):
-    #         submission["id"].append(id)
-    #         submission["target"].append(int(pred))
-    #     submission = pd.DataFrame(submission)
-    #     return submission
     def __init__(
         self,
         data: pd.DataFrame = pd.read_csv("./Modelling/dataset/data/train.csv"),
         test: pd.DataFrame = pd.read_csv("./Modelling/dataset/data/test.csv"),
-        all_words: list = [],
-        tags: list = [],
     ):
         self.data = data
         self.data = self.data.sample(frac=1)
         self.test = test
-        self.X = self.data["text"]
-        self.y = self.data["target"]
-        self.all_words = all_words
-        self.tags = tags
-
-    def tokenize(self, sentence):
-        return nltk.word_tokenize(sentence.lower())
-
-    def stem(self, word):
-        return stemmer.stem(word.lower())
-
-    def words_to_int(self, words, all_words):
-        new_words = []
-        for word in words:
-            new_words.append(self.stem(word))
-        list_of_os = np.zeros(len(all_words))
-        for i in range(len(all_words)):
-            if all_words[i] in new_words:
-                list_of_os[i] = 1.0
-        return list_of_os
-
-    def create_all_words(self):
-        for x_iter, y_iter in tqdm(zip(self.X, self.y)):
-            x_iter = self.tokenize(x_iter)
-            new_x_iter = []
-            for x_iter_i in x_iter:
-                new_x_iter.append(self.stem(x_iter_i))
-            self.all_words.extend(new_x_iter)
-            self.tags.append(y_iter)
-        np.random.shuffle(self.all_words)
-        self.all_words = sorted(set(self.all_words))
-        self.tags = sorted(set(self.tags))
-        return self.all_words, self.tags
-
-    def create(self, test_size=0.0625, shuffle=True):
-        self.create_all_words()
-        self.new_X = []
-        self.new_y = []
-        for X_iter, y_iter in tqdm(zip(self.X, self.y)):
-            self.new_X.append(self.words_to_int(X_iter, self.all_words))
-            self.new_y.append(self.tags.index(y_iter))
-        self.X = np.array(self.new_X)
-        self.y = np.array(self.new_y)
+        self.X = np.array(self.data["text"])
+        self.y = np.array(self.data["target"])
         self.X_train, self.X_test, self.y_train, self.y_test = train_test_split(
-            self.X, self.y, test_size=test_size, shuffle=shuffle
-        )
-        return (
-            self.new_X,
-            self.new_y,
-            self.X,
-            self.y,
-            self.X_train,
-            self.X_test,
-            self.y_train,
-            self.y_test,
-            self.all_words,
+            self.X, self.y, test_size=0.0625, shuffle=True
         )
 
+    def create(self):
+        return self.X_train, self.X_test, self.y_train, self.y_test
+
     def create_test(self):
-        new_test = []
-        for X_iter in tqdm(self.test["text"]):
-            new_test.append(self.words_to_int(X_iter, self.all_words))
-        new_test = torch.from_numpy(np.array(new_test)).to("cpu")
-        return new_test
+        return np.array(self.test["text"])
 
     def create_submission(self, model):
-        model.to("cpu")
-        preds = model(self.create_test().float())
+        preds = model.predict(self.create_test())
+        print(preds)
         ids = self.test["id"]
         submission = {"id": [], "target": []}
         for pred, id in tqdm(zip(preds, ids)):
             submission["id"].append(id)
-            submission["target"].append(int(torch.argmax(pred)))
+            submission["target"].append(int(pred))
         submission = pd.DataFrame(submission)
         return submission
diff --git a/run.py b/run.py
index a0a9f0a..2091eaa 100644
--- a/run.py
+++ b/run.py
@@ -7,25 +7,16 @@ models = [
     "https://tfhub.dev/google/universal-sentence-encoder-multilingual-qa/3",
     "https://tfhub.dev/google/tf2-preview/nnlm-en-dim128/1",
 ]
-models = ["https://tfhub.dev/digitalepidemiologylab/covid-twitter-bert/2"]
+models = ["https://tfhub.dev/google/nnlm-en-dim50/2"]
 for model in models:
     tdl = TensorFlow_Data_Loader()
     (
-        new_X,
-        new_y,
-        X,
-        y,
         X_train,
         X_test,
         y_train,
         y_test,
-        all_words,
     ) = tdl.create()
     tm = TensorFlow_Modelling()
     model = tm.train(X_train, X_test, y_train, y_test, model=model)
     submission = tdl.create_submission(model)
     submission.to_csv(f"./save/{model}-TensorFlow-BaseLine.csv", index=False)
-
-# SKlearn = 77%
-# Pytorch = 60%
-# Ten
diff --git a/wandb/latest-run b/wandb/latest-run
index 5a31bdf..dd2ba88 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20220520_065726-1vuhtw8j
\ No newline at end of file
+run-20220520_075404-2fg179rx
\ No newline at end of file
